beacon_node/network/src/router.rs

```diff
@@ -576,15 +578,12 @@ impl<T: BeaconChainTypes> Router<T> {
 
     /// An error occurred during an RPC request. The state is maintained by the sync manager, so
     /// this function notifies the sync manager of the error.
-    pub fn on_rpc_error(&mut self, peer_id: PeerId, request_id: AppRequestId, error: RPCError) {
-        // Check if the failed RPC belongs to sync
-        if let AppRequestId::Sync(request_id) = request_id {
-            self.send_to_sync(SyncMessage::RpcError {
-                peer_id,
-                request_id,
-                error,
-            });
-        }
+    pub fn on_rpc_error(&mut self, peer_id: PeerId, request_id: SyncRequestId, error: RPCError) {
+        self.send_to_sync(SyncMessage::RpcError {
+            peer_id,
+            request_id,
+            error,
+        });
```

Comment:
This looks like one case where we care if the message is from `Sync` or `Router`. 

The logic here I think has changed. Previously an error that occurred due to a Status message would not get forwarded to sync, but now it does. 

This could have follow on effects, but could be handled in sync I guess

Reply:
thanks Age, handled with c537b6e704c06dcee7052b7660b89c6b4a62d041